themes-default/slim/static/js/store.js

```diff
@@ -244,11 +244,35 @@ const store = new Puex({
             });
         },
         getShows(store, shows) {
-            const { getShow } = this;
-            return shows.forEach(show => getShow(show));
+            const { dispatch } = store;
+
+            // If no shows are provided get all of them
+            if (!shows) {
+                return api.get('/series?limit=1000').then(res => {
+                    const shows = res.data;
+                    return shows.forEach(show => {
+                        store.commit(ADD_SHOW, show);
+                    });
+                });
+            }
+
+            return shows.forEach(show => dispatch('getShow', show));
         },
         testNotifications() {
             return displayNotification('error', 'test', 'test<br><i class="test-class">hello <b>world</b></i><ul><li>item 1</li><li>item 2</li></ul>', 'notification-test--');
+        },
+        setLayout(store, { page, layout }) {
+            const { dispatch } = store;
+
+            return api.patch('config/main', {
+                layout: {
+                    [page]: layout
+                }
+            }).then(setTimeout(() => {
+                dispatch('getConfig');
```

Comment:
1. Why is a timeout needed here?
2. Since we currently refresh the page I suggest commenting out this line.

Reply:
@OmgImAlexis 
I know this is already merged but working on something else I noticed the `.then()` doesn't wait for the API call to return.
Don't know exactly what the issue is, but try looking on the network tab (enable `preserve log` if needed), on the Waterfall section of the network tab, you can see that the request issued by `.then()` doesn't wait for the previous one to finish.